keycloak/openid_client.go

```diff
@@ -80,3 +106,57 @@ func (keycloakClient *KeycloakClient) UpdateOpenidClient(client *OpenidClient) e
 func (keycloakClient *KeycloakClient) DeleteOpenidClient(realmId, id string) error {
 	return keycloakClient.delete(fmt.Sprintf("/realms/%s/clients/%s", realmId, id))
 }
+
+func (keycloakClient *KeycloakClient) GetOpenidClientDefaultScopes(realmId, clientId string) ([]*OpenidClientScope, error) {
+	var scopes []*OpenidClientScope
+
+	err := keycloakClient.get(fmt.Sprintf("/realms/%s/clients/%s/default-client-scopes", realmId, clientId), &scopes)
+	if err != nil {
+		return nil, err
+	}
+
+	return scopes, nil
+}
+
+func (keycloakClient *KeycloakClient) AttachOpenidClientDefaultScopes(realmId, clientId string, scopeNames []interface{}) error {
+	openidClient, err := keycloakClient.GetOpenidClient(realmId, clientId)
```

Comment:
Might be worth adding a length check for the `scopeNames` slice, since there's no reason to make those two API calls below if there are no scopes to attach.

Reply:
Could `scopeNames` be a `[]string`?

Reply:
> Might be worth adding a length check for the `scopeNames` slice, since there's no reason to make those two API calls below if there are no scopes to attach.

I considered this as well but ended up not doing it because there is still value in using this resource to ensure that no default scopes are attached to a client, like so:

```hcl
resource "keycloak_openid_client_default_scopes" "no_default_scopes" {
    realm_id       = "test-realm"
    client_id      = "test-client"
    default_scopes = []
}
```

Since this is a valid use case, then checking the length of the slice before making those API calls would allow a potentially invalid configuration to exist and blow up as soon as one scope is added.

Reply:
> Could `scopeNames` be a `[]string`?

Technically yes, but the return value of `*schema.Set -> List()` is `[]interface{}`, and you need to loop over each item in order to convert that to `[]string` which I have to do anyways in `includeOpenidClientScopesMatchingNames`.

Reply:
Fair enough, but the `keycloak` package is useful independently of the Terraform provider. So from that perspective, it makes sense for it to be `[]string`.

Reply:
Yeah, that's a good point.  You've convinced me, I'll make this change.